### PR TITLE
feat(nat): add a new datasource to query the list of transit ip

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:
Add a new datasource to query the list of transit ip.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Add a new datasource.
1.1 the datasource is used to query the list of transit ip.
1.2 add the datasource documentation (docs/datasource/nat_private_transit_ips).
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (br_transitlist)$ make testacc TEST="./huaweicloud/services/acceptance/nat" TESTARGS="-run TestAccDatasourcePrivateTransitIps_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run TestAccDatasourcePrivateTransitIps_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourcePrivateTransitIps_basic
=== PAUSE TestAccDatasourcePrivateTransitIps_basic
=== CONT  TestAccDatasourcePrivateTransitIps_basic
--- PASS: TestAccDatasourcePrivateTransitIps_basic (97.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       97.671s
```

### CODE REVIEW

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+	listTransitIpsPath += listTransitIpsQueryParams
+
+	listTransitIpsResp, err := pagination.ListAllItems(
+		listTransitIpsClient,
+		"marker",
+		listTransitIpsPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving transit IP")
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]:
```
"error retrieving transit IPs"
```
or
```
"error retrieving transit IP list"
```

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	region := cfg.GetRegion(d)
+
+	// listTransitIps: Query the transit IP list
+	var (
+		listTransitIpsHttpUrl = "v3/{project_id}/private-nat/transit-ips"
+		listTransitIpsProduct = "nat"
+	)
+	listTransitIpsClient, err := cfg.NewServiceClient(listTransitIpsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating transit IP client: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]:
```suggestion
		return diag.Errorf("error creating NAT client: %s", err)
```

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the subnet to which the transit IP belongs.",
+			},
+			"tags": {
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]:
```suggestion
				Computed:    true,
```

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+// API: NAT GET /v3/{project_id}/private-nat/transit-ips
+func DataSourcePrivateTransitIps() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePrivateTransitIpsRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]:
```suggestion
				Optional:    true,
```

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 1 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 2 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 3 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 4 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+)
+
+// API: NAT GET /v3/{project_id}/private-nat/transit-ips
+func DataSourcePrivateTransitIps() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePrivateTransitIpsRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Computed:    true,
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]:
```suggestion
				Optional:    true,
```

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: check again

Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go (hunk 5 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+			"network_interface_id":  utils.PathSearch("network_interface_id", v, nil),
+			"enterprise_project_id": utils.PathSearch("enterprise_project_id", v, nil),
+		})
+	}
+	return rst
+}
+
+func filterListTransitIpsResponseBody(all []interface{}, d *schema.ResourceData) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	tagFilter := d.Get("tags").(map[string]interface{})
```
Comment (ShiChangkuo) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go:219]: In most situation, the `tags` param is not specified, so we can return directly when it is empty.
```suggestion
	tagFilter := d.Get("tags").(map[string]interface{})
	if len(tagFilter) == 0 {
		return all
	}
```

Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_transit_ips.go:219]: done

**Pull request merged**